Content/Items/Lightsaber/LightsaberBase.cs

```diff
@@ -0,0 +1,617 @@
+				oldPositionCollision.RemoveAt(0);
+
+			if (thrown)
+			{
+				for (int i = 0; i < oldPositionDrawing.Count; i++)
+				{
+					oldPositionCollision[i] += Projectile.velocity;
+					oldPositionDrawing[i] += Projectile.velocity;
+					oldSquish[i] = squish;
+				}
+			}
+
+			if (thrown && throwTimer % Projectile.extraUpdates == Projectile.extraUpdates - 1)
+			{
+				for (int i = 0; i < oldPositionDrawing.Count; i++)
+				{
+					oldPositionDrawing[i] += oldScreenPos - Main.screenPosition;
+				}
+			}
+
+			oldScreenPos = Main.screenPosition;
+		}
+
+		public override bool? Colliding(Rectangle projHitbox, Rectangle targetHitbox)
+		{
+			float collisionPoint = 0f;
+
+			for (int i = 0; i < oldPositionCollision.Count; i++)
+			{
+				if (Collision.CheckAABBvLineCollision(targetHitbox.TopLeft(), targetHitbox.Size(), oldPositionCollision[i], GetCollisionPoint(i) + oldPositionCollision[i], 40, ref collisionPoint))
+				{
+					Vector2 dustPos = oldPositionCollision[i] + Vector2.Normalize(GetCollisionPoint(i)) * collisionPoint;
+					Vector2 dustVel = Vector2.Normalize(GetCollisionPoint(i)).RotatedBy(1.57f).RotatedByRandom(0.3f);
+
+					for (int j = 0; j < 11; j++)
+					{
+						float speed = Main.rand.NextFloat();
+						int dir = Main.rand.NextBool() ? 1 : -1;
+						Dust.NewDustPerfect(dustPos, ModContent.DustType<LightsaberGlow>(), dustVel * dir * speed * 5, default, new Color(BladeColor.X * 2, BladeColor.Y * 2, BladeColor.Z * 2), MathHelper.Lerp(0.95f, 0.55f, speed));
+					}
+
+					return true;
+				}
+			}
+
+			return false;
+		}
+
+		public override bool? CanHitNPC(NPC target)
+		{
+			if (hit.Contains(target) || !canHit)
+				return false;
+
+			return base.CanHitNPC(target);
+		}
+
+		public override void OnHitNPC(NPC target, NPC.HitInfo hitinfo, int damageDone)
+		{
+			Projectile.penetrate++;
+			hit.Add(target);
+
+			if (CameraSystem.shake < 20)
+				CameraSystem.shake += 2;
+		}
+
+		public override void ModifyHitNPC(NPC target, ref NPC.HitModifiers modifiers)
+		{
+			modifiers.HitDirectionOverride = Math.Sign(target.Center.X - Owner.Center.X);
+		}
+
+		public override bool PreDraw(ref Color lightColor)
+		{
+			if (hide)
+				return false;
+
+			Texture2D tex = ModContent.Request<Texture2D>(Texture).Value;
+			Texture2D whiteTex = ModContent.Request<Texture2D>(Texture + "_White").Value;
+			Texture2D glowTex = ModContent.Request<Texture2D>(Texture + "_Glow").Value;
+			Texture2D smallGlowTex = ModContent.Request<Texture2D>(Texture + "_SmallGlow").Value;
+
+			//X = pitch, Y = yaw of drawn projectile (regular rotation is roll)
+			Vector2 scaleVec;
+			Effect effect = Filters.Scene["3DSwing"].GetShader().Shader;
+			effect.Parameters["color"].SetValue(Color.White.ToVector4());
+			effect.Parameters["rotation"].SetValue(Projectile.rotation - midRotation);
+
+			for (int i = 0; i < oldPositionDrawing.Count; i++) //disgusting amount of spritebatch restarts but I can't figure out another way to do it
```

Comment:
This behavior will likely need revised. Perhaps we can look into generating a 3D mesh for a trail?

Reply:
I'm not good with 3D things so I can't help with that